src/server/bg_services/namespace_fs_monitor.js

```diff
@@ -0,0 +1,86 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const path = require('path');
+const P = require('../../util/promise');
+const config = require('../../../config');
+const nb_native = require('../../util/nb_native');
+const system_utils = require('../utils/system_utils');
+const dbg = require('../../util/debug_module')(__filename);
+const auth_server = require('../common_services/auth_server');
+const system_store = require('../system_services/system_store').get_instance();
+
+class NamespaceFSMonitor {
```

Comment:
Why is this separated from NamespaceMonitor?

Reply:
It felt more right, As this Monitor is only on the endpoint and not on the core (at list for now).
Another thing is that the `requires` are different and the test also.
The flow is different as we don't even need a connection.

Reply:
To me it makes more sense to have it as another case under the same namespace monitor class, rather than duplicating the entire class and add if(nsfs) and if(!nsfs) in each one respectfully. Sounds like you could extend the existing class with another function and that's it.

Reply:
It is not just namespace vs nsfs, It is where it runs.
We are running it on the endpoint, and if we will run the Namespace_monitor on the endpoint it will run per namespace, even if it is not nsfs.
of course, we can also protect that, but I find it cleaner to divide into 2 files, one that runs in core and checks namespace and the other that's run in the endpoint and checks nsfs. 
WDYT?